### ISSUE TITLE
Tags that are incomplete/ Non-Standard/ Deprecated- Discussion BEFORE requesting new Quest

### ISSUE THREAD

**Issue by chrisdebian** [streetcomplete/StreetComplete#4898]
Hi, everybody.

I'd like to start a discussion about the MANUAL updating of deprecated, missing, incomplete or non-standard tags. Streetcomplete and its contributors are very good at adding missing information, but sometimes information can't be added, because a tag may be deprecated or incomplete. In this instance, the replacement tag is likely to have more tagging options, which would mean that if the tag was updated from the current (deprecated) tag to the new tag, then we could potentially add more data.

Most readers will be aware that automated mass-edits are strongly discouraged, and I understand this; but, I still feel that deprecated/ incomplete tags need to be considered.

A list of deprecated tags can be found [here](https://wiki.openstreetmap.org/wiki/Deprecated_features#Deprecated_features), and this includes Overpass queries.

![image](https://user-images.githubusercontent.com/17935223/227197935-98b3c3d1-2fb3-4908-899a-634b551dfe55.png)

I'm not convinced that this is the definitive list of deprecated tags, as looking at a random area in the UK, I can [see](https://www.openstreetmap.org/edit?relation=146656#map=19/53.47263/-2.18065) (using the ID editor/ click on 'Issues (warning triangle)' on the right hand edge) that other issues exist, such as 'xx has no descriptive tags', 'Marked Crossing has incomplete tags' and 'Petrol Station has non-standard tags'.

![image](https://user-images.githubusercontent.com/17935223/227197223-675fb7b3-77e9-4a3a-ba37-c4342341fa1a.png)

[This page](https://wiki.openstreetmap.org/wiki/Category:Tag_descriptions_with_status_%22deprecated%22) may be helpful.

![image](https://user-images.githubusercontent.com/17935223/227199467-371fe405-c861-4f17-9ebf-0466dca9f9f3.png)

Another resource is KeepRight, which can identify missing tags, misspelled tags and deprecated tags. [Here](https://keepright.at/report_map.php?zoom=14&lat=48.20808&lon=16.37221) is an example; you can select the data you want to see, on the left side of the screen.

![image](https://user-images.githubusercontent.com/17935223/227197531-ccc94a75-ab39-451d-9bf0-4873f1fa760b.png)

I would welcome your thoughts on whether submitting a formal request for a Quest would be sensible.

Many thanks,

Chris
chris_debian
UK

**Comment by matkoniecz** [comment]
In cases where 1:1 replacements are possible it is pointless to waste human time on this. Proper solution is to either

- do not make 1:1 deprecations for the sake of them
- make a bot edit replacing them. 

In case where tag was ambiguous and human help is needed and tag is popular and it can be handled by changing tags then it will likely fulfil requirements from https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md - and some quests are already about this, for example asking about `shop=yes` POIs.

To be more specific: either adding `crossing:markings=yes` to every single `crossing=marked` is a good idea and SC quest can be changed to add also that and bot edit can be run. Or it is a bad idea and iD should not bother people about doing it manually.

Note: if someone will propose such bot edit at talk mailing list (for global ones) or at https://community.openstreetmap.org/ for limited to a specific area (for communities which have it as their primary discussion forum where it is not necessary to discuss bot edit also elsewhere) AND document at wiki as required AND it would be clearly accepted and welcome I likely would be able to run edit itself. But please write first to https://www.openstreetmap.org/message/new/Mateusz%20Konieczny to confirm that bot edit would be easy and viable (not tried this before, hopefully it will work).

Another case: [landuse](https://wiki.openstreetmap.org/wiki/Key:landuse)=[reservoir](https://wiki.openstreetmap.org/wiki/Tag:landuse%3Dreservoir) - here many remaining ones are not actually reservoirs. I considered proposing bot edit here and discovered it to be not viable.

**Comment by matkoniecz** [comment]
Oh, and note that some documentation pages on Wiki are overzealous or simply wrong. Research and/or consulting OSM community is a good idea.

**Comment by matkoniecz** [comment]
> [Here](https://keepright.at/report_map.php?zoom=14&lat=48.20808&lon=16.37221) is an example; you can select the data you want to see, on the left side of the screen.

I looked quickly, it is filled with bogus reports in my area and https://github.com/keepright/keepright is abandoned. 

Osmose is actively developed but for some reason reported false positives are not getting fixed.

I looked some time ago through what JOSM and iD have in validators and added quests for ones where it seemed to be viable and useful. But looking through with https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md  in mind and checking what was proposed already may reveal new ones. Feel free to create issues for them!

**Comment by matkoniecz** [comment]
tl;dr: https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md also applies here, cases where deprecated tag is matching are welcome. But already this lists were checked and obvious ones were implemented.

"No spam" rule means that 1:1 replacements are not eligible.

**Comment by chrisdebian** [comment]
Ok, thank you.

Chris.
